hub/core/partial_reader.py

```diff
@@ -0,0 +1,29 @@
+from typing import Dict
+from hub.core.storage import LRUCache
+
+
+class PartialReader:
+    def __init__(self, cache: LRUCache, path: str, header_offset: int):
+        self.cache = cache
+        self.path = path
+        self.data_fetched: Dict[tuple[int, int], memoryview] = {}
+        self.header_offset = header_offset
+
+    def __getitem__(self, slice_: slice) -> memoryview:
+        start = slice_.start + self.header_offset
+        stop = slice_.stop + self.header_offset
+        step = slice_.step
+        assert start is not None and stop is not None
+        assert step is None or step == 1
+        slice_tuple = (start, stop)
+        if slice_tuple not in self.data_fetched:
+            self.data_fetched[slice_tuple] = memoryview(
+                self.cache.get_bytes(self.path, start, stop)
```

Comment:
Shouldn't there be a minimum fetch size?

Reply:
we're only getting to this part if the number of samples is less than 20. So that will automatically take care of minimum fetch size